### PR TITLE
PrioritizedQueue optimization and some trivial optimization

### PR DESCRIPTION
**Pull request by XinzeChi**

### CODE REVIEW

**Comment by XinzeChi** [comment]
test this please

File: src/common/PrioritizedQueue.h [hunk 1]
```diff
@@ -115,19 +120,35 @@ class PrioritizedQueue {
...
       q[cl].push_back(std::make_pair(cost, item));
-      if (cur == q.end())
+      if (cur == q.end()) {
 	cur = q.begin();
+        cur_key = cur->first;
+      }
+      else if (q.bucket_count() != bucket_count) {
+        cur = q.find(cur_key);
+        assert(cur != q.end());
+        cur_key = cur->first;
```
Comment (mslovy) [inline src/common/PrioritizedQueue.h]: cur_key is already the key for cur?
Also, even if the bucket_count changes, the iterator is still valid and points to the original item.

**Comment by mslovy** [comment]
@XinzeChi  is it better to initialize the vector size to CEPH_MSG_PRIO_LOW for queue and CEPH_MSG_PRIO_HIGHEST - CEPH_MSG_PRIO_LOW for high_queue in constructor?

File: src/common/PrioritizedQueue.h (hunk 1 above)
Comment (XinzeChi) [inline src/common/PrioritizedQueue.h]: Maybe, but I am not so confident that whether the implement of unordered_map are the same in different platform. I think I would be better to treat it as invalid when bucket_count changes?

**Comment by XinzeChi** [comment]
So we should pass the CEPH_MSG_PRIO_HIGHEST and CEPH_MSG_PRIO_LOW  to PrioritizedQueue as parameter? It looks a little tricky?

File: src/common/PrioritizedQueue.h (hunk 1 above)
Comment (mslovy) [inline src/common/PrioritizedQueue.h]: Yeah, so just keep the fail assertion there for confidence?  But at least,  cur_key = cur->first is redundant, I think.

**Comment by markhpc** [comment]
Hi Xinzechi,

It appears there a couple of different distinct optimizations in this PR.  Is that correct?  It would be nice if we could split these up into separate PRs and for non-trivial changes, see some kind of test result showing an improvement.  Thanks!

**Comment by athanatos** [comment]
I'm closing this one for now.  It needs to be rebased on the version.  Also, I'd want to see numbers demonstrating a performance improvement.  Finally, WPQ is probably the way we are going to go in the future, so it's not clear that it's worth improving PrioritizedQueue.

**Pull request closed without merging**